sentry-apache-http-client-5/src/main/java/io/sentry/transport/apache/ApacheHttpClientTransport.java

```diff
@@ -71,68 +85,93 @@ public void send(final @NotNull SentryEnvelope envelope, final @Nullable Map<Str
       final SentryEnvelope filteredEnvelope = rateLimiter.filter(envelope, sentrySdkHint);
 
       if (filteredEnvelope != null) {
-        currentlyRunning.increment();
-
-        try (final ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
-            final GZIPOutputStream gzip = new GZIPOutputStream(outputStream)) {
-          options.getSerializer().serialize(filteredEnvelope, gzip);
-
-          final SimpleHttpRequest request =
-              SimpleHttpRequests.post(requestDetails.getUrl().toString());
-          request.setBody(
-              outputStream.toByteArray(), ContentType.create("application/x-sentry-envelope"));
-          request.setHeader("Content-Encoding", "gzip");
-          request.setHeader("Accept", "application/json");
-
-          for (Map.Entry<String, String> header : requestDetails.getHeaders().entrySet()) {
-            request.setHeader(header.getKey(), header.getValue());
-          }
+        final SentryEnvelope envelopeWithClientReport =
+            clientReportRecorder.attachReportToEnvelope(filteredEnvelope, options);
+
+        if (envelopeWithClientReport != null) {
+          currentlyRunning.increment();
+
+          try (final ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+              final GZIPOutputStream gzip = new GZIPOutputStream(outputStream)) {
+            options.getSerializer().serialize(envelopeWithClientReport, gzip);
+
+            final SimpleHttpRequest request =
+                SimpleHttpRequests.post(requestDetails.getUrl().toString());
+            request.setBody(
+                outputStream.toByteArray(), ContentType.create("application/x-sentry-envelope"));
+            request.setHeader("Content-Encoding", "gzip");
+            request.setHeader("Accept", "application/json");
+
+            for (Map.Entry<String, String> header : requestDetails.getHeaders().entrySet()) {
+              request.setHeader(header.getKey(), header.getValue());
+            }
+
+            if (options.getLogger().isEnabled(DEBUG)) {
+              options
+                  .getLogger()
+                  .log(DEBUG, "Currently running %d requests", currentlyRunning.getCount());
+            }
+
+            httpclient.execute(
+                request,
+                new FutureCallback<SimpleHttpResponse>() {
+                  @Override
+                  public void completed(SimpleHttpResponse response) {
+                    if (response.getCode() != 200) {
+                      options
+                          .getLogger()
+                          .log(ERROR, "Request failed, API returned %s", response.getCode());
+
+                      if (!(sentrySdkHint instanceof Retryable)) {
+                        if (response.getCode() >= 500) {
```

Comment:
I believe everything but 200 would be a network error, if our API always results in 200 anyway.

Reply:
The idea is to not handle 429 as network error. I can also make this more specific and only exclude 429, then everything >= 400 except 429 would be counted as network error.

Reply:
4xx makes sense since it's a client error, there's no reason why the server would not have reported client reports unless they use the wrong HTTP error code.
The question is, if the server throws 5xx, do we report client reports or not? worth asking on client-infra.

Reply:
- [ ] ask on client infra, which status codes should increase client reports counts

Reply:
Asked on client infra channel, outcome here: https://github.com/getsentry/develop/pull/554
> SDKs can put a simple optional check for HTTP status codes in place where any code `>= 400` except `429` will be recorded as `network_error`.

Reply:
- [ ] Put HTTP status code handling in check for >= 400 except 429

Reply:
I created an issue on Cocoa for that https://github.com/getsentry/sentry-cocoa/issues/1753